modules/schema/src/main/java/org/apache/ignite/internal/schema/registry/SchemaRegistryImpl.java

```diff
@@ -145,13 +146,17 @@ public int lastSchemaVersion() {
     public Row resolve(BinaryRow row, int targetSchemaVersion) {
         SchemaDescriptor targetSchema = schema(targetSchemaVersion);
 
+        if (targetSchema == null) {
+            throw new SchemaRegistryException("No schema found: schemaVersion=" + targetSchemaVersion);
+        }
+
         return resolveInternal(row, targetSchema, false);
     }
 
     /** {@inheritDoc} */
     @Override
     public Row resolve(BinaryRow row, SchemaDescriptor schemaDescriptor) {
-        return resolveInternal(row, schemaDescriptor, false);
+        return resolveInternal(row, Objects.requireNonNull(schemaDescriptor), false);
```

Comment:
Maybe `assert`? Why don’t we do checks for the `row` too? 
I suggest not checking here, this is an internal API in which we have some rules.